stock_inventory_revaluation/models/stock_inventory_revaluation.py

```diff
@@ -0,0 +1,407 @@
+        datas = self.env['product.template'].get_product_accounts(
+            self.product_template_id.id)
+        self.move_id = self.env['account.move'].create(move_data).id
+        move_line_obj = self.env['account.move.line']
+
+        if not self.decrease_account_id or not self.increase_account_id:
+            raise UserError(_("Please add a  Increase Account and "
+                              "a Decrease Account."))
+
+        for prod_variant in self.product_template_id.product_variant_ids:
+                qty = prod_variant.qty_available
+                if qty:
+                    if amount_diff > 0:
+                        debit_account_id = self.decrease_account_id.id
+                        credit_account_id = \
+                            datas['property_stock_valuation_account_id']
+                    else:
+                        debit_account_id = \
+                            datas['property_stock_valuation_account_id']
+                        credit_account_id = self.increase_account_id.id
+                    move_line_data = self._prepare_debit_move_line_data(
+                        abs(amount_diff), debit_account_id, prod_variant.id)
+                    move_line_obj.create(move_line_data)
+                    move_line_data = self._prepare_credit_move_line_data(
+                        abs(amount_diff), credit_account_id, prod_variant.id)
+                    move_line_obj.create(move_line_data)
+                    if self.move_id.journal_id.entry_posted:
+                        self.move_id.post()
+
+    @api.one
+    def post(self):
+
+        amount_diff = 0.0
+        if self.product_template_id.cost_method == 'real':
+            for reval_quant in self.reval_quant_ids:
+                amount_diff += reval_quant.get_total_value()
+                reval_quant.write_new_cost()
+            if amount_diff == 0.0:
+                return True
+        else:
+            if self.product_template_id.cost_method in ['standard', 'average']:
+
+                if self.revaluation_type == 'price_change':
+                    diff = self.current_cost - self.new_cost
+                    amount_diff = self.qty_available * diff
+                else:
+                    amount_diff = self.current_value - self.new_value
+                    if self.new_value < 0:
+                        raise UserError(_("The new value for product %s "
+                                          "cannot be negative"
+                                          % self.product_template_id.name))
+                    if self.qty_available <= 0.0:
+                        raise UserError(
+                            _("Cannot do an inventory value change if the "
+                              "quantity available for product %s "
+                              "is 0 or negative" %
+                              self.product_template_id.name))
+
+                if self.revaluation_type == 'price_change':
```

Comment:
Because you're testing different thing. One is quantity, another is value (amount)